### ISSUE TITLE
Is allreduce needed for custom loss function for tf.keras

### ISSUE THREAD

**Issue by dktemp888** [horovod/horovod#1848]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet): tf.keras
2. Framework version: TF 2.1.0
3. Horovod version: 0.19.1
4. MPI version: 4.0.1
5. CUDA version: n.a.
6. NCCL version: n.a.
7. Python version: 3.7.5
8. OS and version: CentOS 7
9. GCC version: 

**Checklist:**
1. Did you search issues to find if somebody asked this question before? Yes
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
Please ask your question here.

Assume I have below custom loss function for single instance my tf.keras model:
```
def my_loss(y_true, y_pred):  
    d = tf.abs(y_true - y_pred)
    cost = tf.reduce_mean(d)
    return cost
...

model.compile(loss=my_loss, optimizer=opt)
```
When using `horovod.tf.keras` for distributed training:
1. should I use `hvd.allreduce()` for example `return hvd.allreduce(cost)` to have my_loss taken care of loss aggregation among all ranks? Is it even possible to `all_reduce` cost here?
2. what does `hvd.DistributedOptimizer` wrapper and `hvd.callbacks.BroadcastGlobalVariablesCallback(0)` do if I don't use own loss function?

I wonder if someone can explain it in more details. In `pytorch.distributed` package, training loss from each rank should be all_reduce first and then back propagate. Here i am a bit confused in who does what.

Thanks
